### PR TITLE
480 separate fe be code

### PR DESCRIPTION
**Pull request by shailee-m**
Separated frontend and backend code:
1. Made one method that renders the charts,
2. Added frontend tests for collection hooks

### CODE REVIEW

**Comment by shailee-m** [comment]
@brylie still resolving some code climate issues but PR is ready to be reviewed

**Comment by brylie** [comment]
K. I will review ASAP.

File: app/both/collections/activities.js [hunk 1]
```diff
@@ -165,3 +165,6 @@ Activities.after.remove(function (userId, activity) {
     entityId: activity._id,
   })
 });
+
+
+module.exports=Activities
```
Comment (brylie) [inline app/both/collections/activities.js]: Let's use [ES2015 export syntax](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export).

File: app/both/collections/activityTypes.js [hunk 2]
```diff
@@ -73,3 +73,4 @@ ActivityTypes.after.remove(function (userId, activityType) {
     entityId: activityType._id,
   })
 });
+module.exports = ActivityTypes
```
Comment (brylie) [inline app/both/collections/activityTypes.js]: Let's use [ES2015 export syntax](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export).

File: app/both/collections/groups.js [hunk 3]
```diff
@@ -80,3 +80,5 @@ Groups.after.remove(function (userId, group) {
     entityId: group._id,
   })
 });
+
+module.exports = Groups;
```
Comment (brylie) [inline app/both/collections/groups.js]: Let's use [ES2015 export syntax](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export).

File: app/both/collections/homes.js [hunk 4]
```diff
@@ -65,3 +65,5 @@ Homes.after.remove(function (userId, home) {
     entityId: home._id,
   })
 });
+
+module.exports = Homes
```
Comment (brylie) [inline app/both/collections/homes.js]: Let's use [ES2015 export syntax](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export).

File: app/both/collections/permissions.js [hunk 5]
```diff
@@ -102,3 +102,4 @@ Permissions.allow({
     }
   }
 });
+module.exports = Permissions
```
Comment (brylie) [inline app/both/collections/permissions.js]: Let's use [ES2015 export syntax](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export).

File: app/both/collections/residencies.js [hunk 6]
```diff
@@ -192,4 +192,5 @@ function hasOtherActiveResidencies(residentId, residencyId, moveOut, moveIn) {
       }
     );
   }
-}
\ No newline at end of file
+}
+module.exports = Residencies
```
Comment (brylie) [inline app/both/collections/residencies.js]: Let's use [ES2015 export syntax](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export).

File: app/both/collections/residents.js [hunk 7]
```diff
@@ -11,59 +11,58 @@ var ResidentsSchema = new SimpleSchema({
     type: String,
     autoform: {
       readonly() {
-        return !isUserAdmin() ? !isEdit(this) : false
-      }
+        return !isUserAdmin() ? !isEdit(this) : false;
```
Comment (brylie) [inline app/both/collections/residents.js]: Add a comment to explain the logic here, as it is somewhat unclear.

File: app/both/collections/residents.js [hunk 8]
```diff
@@ -11,59 +11,58 @@ var ResidentsSchema = new SimpleSchema({
...
+      },
     },
     custom() {
       return !isUserAdmin()
         ? isEdit(this)
-          ? "notAllowed"
+          ? 'notAllowed'
```
Comment (brylie) [inline app/both/collections/residents.js]: This is somewhat dense. Let's make it a bit more verbose to improve readability.

File: app/both/collections/residents.js [hunk 9]
```diff
@@ -11,59 +11,58 @@ var ResidentsSchema = new SimpleSchema({
...
```
Comment (brylie) [inline app/both/collections/residents.js]: Again, this type of code is difficult to read. Try to improve readability, at the expense of having a couple more lines of code.

File: app/both/collections/residents.js [hunk 10]
```diff
@@ -73,98 +72,120 @@ Residents.helpers({
...
-  remove: function (userId, doc) {
-    const schemaType = "resident";
-    const action = "remove";
+  remove: function(userId, doc) {
+    const schemaType = 'resident';
+    const action = 'remove';
     const residentId = doc._id;
-    const activeResidency = Residencies.findOne({ $and: [{ residentId }, { moveOut: { $exists: false } }] });
-    doc.homeId = activeResidency.homeId
+    const activeResidency = Residencies.findOne({
```
Comment (brylie) [inline app/both/collections/residents.js]: Give this code some breathing room with line breaks and comments for the dense lines.

File: app/both/collections/settings.js [hunk 11]
```diff
@@ -61,3 +60,5 @@ Settings.after.remove(function (userId, setting) {
     entityId: setting._id,
   })
 });
+
+module.exports = Settings
```
Comment (brylie) [inline app/both/collections/settings.js]: Let's use [ES2015 export syntax](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export).

File: app/both/collections/userEventLog.js [hunk 12]
```diff
@@ -70,3 +70,5 @@ UserEventLog.allow({
     }
   }
 });
+
+module.exports = UserEventLog
```
Comment (brylie) [inline app/both/collections/userEventLog.js]: Let's use [ES2015 export syntax](https://developer.mozilla.org/en-US/docs/web/javascript/reference/statements/export).

**Comment by brylie** [comment]
Overall, this looks good. Remember that we don't need to solve things in as few characters as possible, so try to avoid really dense or nested conditional shorthand. Also, give the code some room to breath using line breaks, in a similar manner to how paragraphs are used in written text.

**Comment by shailee-m** [comment]
@brylie Made all PR suggestions. Also, resolved the code climate checks. the remaining are in the test cases about duplicating blocks which I think are bound to repeat.

**Comment by brylie** [comment]
OK, we can almost merge this it seems :-) There are just some new conflicts preventing rebase. Please resolve.

**Comment by shailee-m** [comment]
> OK, we can almost merge this it seems :-) There are just some new conflicts preventing rebase. Please resolve.

done

**Comment by brylie** [comment]
Great, I will consider merging this ASAP. I have some open concerns about our release process and wonder where this should fit in the release schedule.

We also need to figure out the next steps on the roadmap since the FE and BE can now evolve independently.

**Comment by shailee-m** [comment]
> Great, I will consider merging this ASAP. I have some open concerns about our release process and wonder where this should fit in the release schedule.
> 
> We also need to figure out the next steps on the roadmap since the FE and BE can now evolve independently.

@brylie  Do you want to discuss about the further developments? I am out of tasks now so may be I can start working on something. I will be available from Monday.

**Comment by brylie** [comment]
Sure. I will check with Marjo about our development priorities.

**Comment by shailee-m** [comment]
> Sure. I will check with Marjo about our development priorities.

Hi any updates?

**Pull request merged**